### ISSUE TITLE
 Unable to serialize to JSON: Object of type coroutine is not JSON serializable

### ISSUE THREAD

**Issue by mrnix1001** [custom-components/alexa_media_player#403]
**Describe the bug**
When using my alexa to play a radio station the following error will happen almost immediately:
`
2019-10-01 10:54:23 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection.139716555925904] Unable to serialize to JSON: Object of type coroutine is
not JSON serializable
{'id': 2, 'type': 'event', 'event': <Event state_changed[L]: entity_id=media_player.office_2, old_state=<state media_player.office_2=paused; volume_level=0.47, is_volume
_muted=False, media_content_type=music, media_duration=0, media_position=4172, media_position_updated_at=2019-10-01T10:39:21.889868-04:00, media_title=WUNC, media_artist
=North Carolina Public Radio, source=Polk SignaS1, source_list=['Local Speaker'], available=True, last_called=True, friendly_name=Office, entity_picture=/api/media_playe
r_proxy/media_player.office_2?token=11bcad7bd06b65c2321a4b44dbe709f06338c50f91586c832701a41a4d581919&cache=010aed74b3a3a135, supported_features=56253 @ 2019-10-01T10:39:
21.890159-04:00>, new_state=<state media_player.office_2=paused; volume_level=0.47, is_volume_muted=False, media_content_type=music, media_duration=0, media_position=417
2, media_position_updated_at=2019-10-01T10:39:21.889868-04:00, media_title=WUNC, media_artist=North Carolina Public Radio, source=<coroutine object AlexaClient._get_sour
ce at 0x7f124bb7e7a0>, source_list=<coroutine object AlexaClient._get_source_list at 0x7f124b424d40>, available=True, last_called=True, friendly_name=Office, entity_pict
ure=/api/media_player_proxy/media_player.office_2?token=11bcad7bd06b65c2321a4b44dbe709f06338c50f91586c832701a41a4d581919&cache=010aed74b3a3a135, supported_features=56253
 @ 2019-10-01T10:39:21.890159-04:00>>}
`

This causes both HA and node red to stop functioning. Only a restart will correct it.

**To Reproduce**
"Alexa, connect to the speakers. Alexa, play NPR".

**Expected behavior**
Not crash HA :)

**Screenshots**

**System details**
 - Home-assistant 0.99.3
 - Hassio (Yes/No): Yes
 - alexa_media  2.0.0, updated to 2.3.0
 - alexapy 1.0.0

**Logs**
2019-10-01 10:54:23 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/config/custom_components/alexa_media/media_player.py", line 199, in _handle_event
    await self.async_schedule_update_ha_state()
TypeError: object NoneType can't be used in 'await' expression
2019-10-01 10:54:23 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection.139716533505936] Unable to serialize to JSON: Object of type coroutine is
not JSON serializable

**Additional context**

**Comment by alandtse** [comment]
You have a really old version of alexapy that you manually installed. Please find and delete it.

**Comment by mrnix1001** [comment]
> You have a really old version of alexapy that you manually installed. Please find and delete it.

It's actually 1.3.1, but I have never installed this manually, I went thru HACs.

**Comment by alandtse** [comment]
Please uninstall through HACS and then confirm the alexa_media directory is deleted from the `/config/custom_components/` directory before reinstalling. The error you are reporting was reported a few months ago #302 and fixed a while back. The logs are showing you have an old version.

If it persists after that, please turn on logging for alexa_media and then provide them for review. The wiki explains how in the FAQ section.

**Comment by mrnix1001** [comment]
All right. I uninstalled, verified that custom_components/alexa_media was not present, reinstalled and restarted HA. During my restart:

alexa_media
Version: 2.3.0
This is a custom component
If you have any issues with this you need to open an issue here:
https://github.com/custom-components/alexa_media_player/issues
-------------------------------------------------------------------

2019-10-02 17:33:34 INFO (MainThread) [custom_components.alexa_media] Loaded alexapy==1.3.1

I will test to see if the issue persists and report back.

Thank you so very much for your help! The Alexa TTS is central to more automations than I'd like to admit!

**Comment by mrnix1001** [comment]
Well, it didn't generate any errors in the first 5 minutes (which it did previously) so I'm leaning towards this being resolved. I don't know how I got a mismatch of versions when everything I did was thru HACs, but the old "delete and reinstall" seemed, at the moment, to have resolved it. I would like this ticket to remain open for at least a day until I do my standard routine that always triggered the issue. If by tomorrow evening I haven't experienced the problem I will close it.

Again, thank you for your help!
